browsers/firefox_android.json

```diff
@@ -54,237 +54,237 @@
         },
         "14": {
           "release_date": "2012-06-26",
-          "release_notes": "https://developer.mozilla.org/Firefox/Releases/14",
+          "release_notes": "https://www.mozilla.org/firefox/android/14.0beta/releasenotes/",
```

Comment:
This shoudn’t point to the beta release notes, but to the [normal release notes](https://www.mozilla.org/firefox/android/14.0/releasenotes/).